ucm2/dsp/intel-dsp-init.conf

```diff
@@ -0,0 +1,111 @@
+# the kcontrols initial values, which will be set by `alsactl init`
+
+If.pga1p {
+	Condition {
+		Type ControlExists
+		Control "name='PGA1.0 1 Master Playback Volume'"
+	}
+	True {
+		BootSequence [
+			cset "name='PGA1.0 1 Master Playback Volume' 32"
+		]
+	}
+}
+
+If.pga2p {
+	Condition {
+		Type ControlExists
+		Control "name='PGA2.0 2 Master Playback Volume'"
+	}
+	True {
+		BootSequence [
+			cset "name='PGA2.0 2 Master Playback Volume' 32"
+		]
+	}
+}
+
+If.pga3p {
+	Condition {
+		Type ControlExists
+		Control "name='PGA3.0 3 Master Playback Volume'"
+	}
+	True {
+		BootSequence [
+			cset "name='PGA3.0 3 Master Playback Volume' 32"
+		]
+	}
+}
+
+# When the platforms support 4 channels speaker playback with RT1308,
+# PGA3.1 3 is used
+If.pga3_1p {
+	Condition {
+		Type ControlExists
+		Control "name='PGA3.1 3 Master Playback Volume'"
+	}
+	True {
+		BootSequence [
+			cset "name='PGA3.1 3 Master Playback Volume' 32"
+		]
+	}
+}
+
+If.pga4p {
+	Condition {
+		Type ControlExists
+		Control "name='PGA4.0 4 Master Playback Volume'"
+	}
+	True {
+		BootSequence [
+			cset "name='PGA4.0 4 Master Playback Volume' 32"
+		]
+	}
+}
+
+If.pga5p {
+	Condition {
+		Type ControlExists
+		Control "name='PGA5.0 5 Master Playback Volume'"
+	}
+	True {
+		BootSequence [
+			cset "name='PGA5.0 5 Master Playback Volume' 32"
+		]
+	}
+}
+
+If.pga6p {
+	Condition {
+		Type ControlExists
+		Control "name='PGA6.0 6 Master Playback Volume'"
+	}
+	True {
+		BootSequence [
+			cset "name='PGA6.0 6 Master Playback Volume' 32"
+		]
+	}
+}
+
+If.pga7p {
+	Condition {
+		Type ControlExists
+		Control "name='PGA7.0 7 Master Playback Volume'"
+	}
+	True {
+		BootSequence [
+			cset "name='PGA7.0 7 Master Playback Volume' 32"
+		]
+	}
+}
+
+If.pga8p {
+	Condition {
+		Type ControlExists
+		Control "name='PGA8.0 8 Master Playback Volume'"
+	}
+	True {
+		BootSequence [
+			cset "name='PGA8.0 8 Master Playback Volume' 32"
```

Comment:
Why do we need to initialize all this? Isn't 32 == 0dB == the default?

Reply:
This is because we should let user have a way to reset to the default value in case the kcontrols value is wrongly set by mistake.

Reply:
We don't do this for codecs, which have a lot more controls that we do, so why would we do it for PGAs only? This seems both overkill and incomplete. if people muck with alsamixer, then there's no much we can do to prevent conflicts.

Reply:
User may change the volumes by mistake and he never knows how to recover them. In this case, if we offer a tune-key solution setting, it will be much easier for user. And also we offer some codec specific settings in such as rt711/init.conf.

Reply:
@libinyang you're not getting my point. There are tons of codec registers and we ONLY set the values that are not the default. Why would do anything different here? If the user changes the value manually, well that's too bad.

Reply:
@plbossart @kv2019i @perexg This should be a question: Do we need to handle the user's mistake? It's really bad that user changes the setting wrongly and we did meet this issue before when we supported the customers. I prefer to try to handle the user mistake as it doesn't take too much effort for us to handle such issue. If we don't handle this issue, user may never make the sound back if he don't know how to set the kcontrols correctly. Even he reboots the system, the sound won't be back because alsactl will always restore the wrong setting.

Reply:
My opinion is that we should allow to reset the values to a known and working state. It will make the use more robust. When we implemented the first ALSA drivers, we had everything muted (off) and the user space was responsible to do the initial settings.

Reply:
we already set the values to a known and working state at the driver level.

What happens in userspace is rather unclear to me, e.g. if you have a UCM file shouldn't it discard any changes made during the last session? Why would the UCM file need to restore the default values that are already set by the driver? it'd seem simpler to me to discard changes made by userspace so that driver default + UCM setup provides a reproducible configuration at each card creation.

Reply:
I think alsactl will restore the previous setting because 'working state' may not be enough for users. The output devices are different totally. Some users may use headphone while others may use AV receiver. So the initial volume may be horribly too loud for some users and may be too low for other users. And another reason I can think is that users may hope OS remember which output is the default output in next boot up. I checked Windows 7, it will also restore the user setting when boot up.

Reply:
@libinyang you are just confusing the heck out of me. The BootSequence will precisely override user settings that might have been set with amixer. We should NOT change the PGA volumes in general anyways, by default it's better to keep the full dynamic range in the DSP and change the volume at the codec level.
Also the topology does define some defaults, so you are just duplicating the work already done by the topology init.

This is really really confusing.

Reply:
The BootSequence takes effect only when `alsactl init` runs. And `alsactl init` is supposed to run when the first boot of the new machine for the initial value setting. `alsactl init` won't be run at the second, third.. boot. Also user can manually run `alsactl init` if he found his machine doesn't have sound any more even after reboot.

So the sequence is:
1. For the first time boot up: driver init -> `alsactl init`
2. For other boots up except the first time: driver init -> `alsactl restore`
3. And at anytime user found there is no sound, user can run `alsactl init` manually (in this case, we need the correct setting of PGA and codec).

Reply:
Hmm, this is a good discussion. I'm afraid step (3) can be more complicated as audio servers (like Pulseaudio) also store state. This doesn't mean we should not provide means to init the state for ALSA level, but just to highlight the recovery may need more steps from end-user perspective. So requiring user to reboot (or reload the driver), in addition to "alsactl init" (and resetting PA state) does not seem unreasonable. 

I was initially thinking that the PGA defaults are fairly harmless and they do add value (recovery if user or buggy application messing with ALSA controls directly), but I think @plbossart raises valid points. Maybe it's not worth the maintenance overhead and we should follow a more conservative approach, where we only provide settings which differ from driver defaults, in BootSequence. Digital/DSP amplify/gain stages typically have 0dB as the default state and this is what we are following for SOF PGAs as well. On some platforms we might want to boost e.g. the DMIC more and this could go to BootSequence, but this would be a non-0dB setting. And I think UCM is a better place for this than topology (UCM can be overridden with longname lookup).